gremlin/ui/device.py

```diff
@@ -283,10 +381,15 @@ def inputIdentifier(self, index: int) -> InputIdentifier:
         return identifier
 
     def _name(self, identifier: Tuple[InputType, int]) -> str:
-        return "{} {:d}".format(
+        input_name = "{} {:d}".format(
             InputType.to_string(identifier[0]).capitalize(),
             identifier[1]
-        )
+            )
+
+        if self.device_mapping is not None:
+            return DeviceDatabase().input_name(input_name, self.device_mapping)
```

Comment:
Since we get a mapping might be nicer for this to be an actual object we can just use rather than having to go through the database each time as it's a bit weird that we give that singleton it's own data back.